### PR TITLE
3025 As an STT user, I need an accurate error report when I space-fill `COUNTY_FIPS_CODE`

### LINKED ISSUE #3025: As an STT user, I need an accurate error report when I space-fill `COUNTY_FIPS_CODE`
**Description:**
_Provide a brief background and justification for this issue_

When STTs submit records with space-filled value for `COUNTY_FIPS_CODE`, the following error appears in the logs:
```shell=
   2024-06-05T14:04:48.89-0400 [APP/PROC/WEB/0] ERR 2024-06-05 18:04:48,898 DEBUG fields.py::parse_value:L45 :  Field: 'COUNTY_FIPS_CODE' at position: [19, 22) is empty.
   2024-06-05T14:04:48.89-0400 [APP/PROC/WEB/0] ERR 2024-06-05 18:04:48,898 ERROR parse.py::manager_parse_line:L370 :  'NoneType' object has no attribute 'lstrip'
```

The error report associated with files like this yield incomplete and incorrect error messages (see below). In particular, 

- the report does not raise a cat 2 (out-of-range) error associated the `COUNTY_FIPS_CODE`
- the report raises cat 4 errors that are not correct (perhaps due to the fact that the T1 record with the space-filled county fips code is not parsed?)

3025p1

STT users are unable to make corrections based on this feedback. 

**Acceptance Criteria:** 
_Create a list of functional outcomes that must be achieved to complete this issue_

- [x] space-filled `COUNTY_FIPS_CODE` raises cat 2 error message in error report. 
- [x] T1 record associated with space-filled `COUNTY_FIPS_CODE`  is still parsed
- [x] cat 4 error message are not raised due to the space-filled `COUNTY_FIPS_CODE` 
- [ ] Testing Checklist has been run and all tests pass
- [ ] README is updated, if necessary

**Tasks:**
_Create a list of granular, specific work items that must be completed to deliver the desired outcomes of this issue_ 

- [x] review use of lstrip with `COUNTY_FIPS_CODE` 
- [ ] resolve issue described
- [ ] tbd
- [ ] Run Testing Checklist and confirm all tests pass

**Notes:**
_Add additional useful information, such as related issues and functionality that isn't covered by this specific issue, and other considerations that will be helpful for anyone reading this_ 

- _Note 1_
- _Note 2_
- _Note 3_

**Supporting Documentation:**
_Please include any relevant log snippets/files/screen shots_

- test file: 
[lstrip-related-records_anon.txt](https://github.com/user-attachments/files/15633171/lstrip-related-records_anon.txt)

- error report associated with test file: 
[2024-Q2-Active Case Data_testfile.xlsx](https://github.com/user-attachments/files/15633355/2024-Q2-Active.Case.Data_testfile.xlsx)

**Open Questions:**
_Please include any questions or decisions that must be made before beginning work or to confidently call this issue complete_ 

- _Open Question 1_
- _Open Question 2_

### PR DESCRIPTION
**Pull request by raftmsohani**
## Summary of Changes

Pull request closes #3025. 
This PR fixes the issue for when report is space filled for COUNTY_FIPS_CODE. The Since the field is space filled it is considered as None in which case throws error for applying ```lstrip``` function to None value. 

## How to Test

Run the following commands to start the 
```
cd tdrs-frontend && docker-compose -f docker-compose.yml -f docker-compose.local.yml up -d
cd tdrs-backend && docker-compose -f docker-compose.yml -f docker-compose.local.yml up -d
```

1. Open http://localhost:3000/ and sign in.
1. Proceed with functional tests as described herein.
1. Submit test file that has space filled COUNTY_FIPS_CODE (see test file in ticket)
1. Make sure the error is clear for not providing value for COUNTY_FIPS_CODE

## Deliverables
_More details on how deliverables herein are assessed included [here](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverables)._

### [Deliverable 1: Accepted Features](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-1-Accepted-Features)

Checklist of ACs:
- [x] space-filled `COUNTY_FIPS_CODE` raises cat 2 error message in error report. 
- [x] T1/M1/T2/M2 record associated with space-filled `COUNTY_FIPS_CODE`  is still parsed
- [x] cat 4 error message are not raised due to the space-filled `COUNTY_FIPS_CODE` 
+ [x] **`lfrohlich`** and/or **`adpennington`**  confirmed that ACs are met.

### [Deliverable 2: Tested Code](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-2-Tested-Code)

+ Are all areas of code introduced in this PR meaningfully tested?
  + [x] If this PR introduces backend code changes, are they meaningfully tested?
  + [ ] If this PR introduces frontend code changes, are they meaningfully tested?
+ Are code coverage minimums met?
  + [ ] Frontend coverage: [_insert coverage %_] (see `CodeCov Report` comment in PR)
  + [x] Backend coverage: [_insert coverage %_] (see `CodeCov Report` comment in PR)

### [Deliverable 3: Properly Styled Code](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-3-Properly-Styled-Code)

+ [x] Are backend code style checks passing on CircleCI?
+ [ ] Are frontend code style checks passing on CircleCI?
+ [ ] Are code maintainability principles being followed?

### [Deliverable 4: Accessible](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-4-Accessibility)

+ [ ] Does this PR complete the epic? 
+ [ ] Are links included to any other gov-approved PRs associated with epic?
+ [ ] Does PR include documentation for Raft's a11y review? 
+ [ ] Did automated and manual testing with `iamjolly` and `ttran-hub` using Accessibility Insights reveal any errors introduced in this PR?

### [Deliverable 5: Deployed](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-5-Deployed)

+ [x] Was the code successfully deployed via automated CircleCI process to development on Cloud.gov?

### [Deliverable 6: Documented](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-6-Code-documentation)

+ [x] Does this PR provide background for why coding decisions were made?
+ [ ] If this PR introduces backend code, is that code easy to understand and sufficiently documented, both inline and overall?
+ [ ] If this PR introduces frontend code, is that code easy to understand and sufficiently documented, both inline and overall?
+ [ ] If this PR introduces dependencies, are their licenses documented?
+ [x] Can reviewer explain and take ownership of these elements presented in this code review?

### [Deliverable 7: Secure](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-7-Secure)

+ [x] Does the OWASP Scan pass on CircleCI?
+ [ ] Do manual code review and manual testing detect any new security issues?
+ [ ] If new issues detected, is investigation and/or remediation plan documented? 

### [Deliverable 8: User Research](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-8-User-Research)

Research product(s) clearly articulate(s):
+ [ ] the purpose of the research
+ [ ] methods used to conduct the research 
+ [ ] who participated in the research
+ [ ] what was tested and how
+ [ ] impact of research on TDP
+ [ ] (_if applicable_) final design mockups produced for TDP development

### CODE REVIEW

File: tdrs-backend/tdpservice/parsers/transforms.py [hunk 1]
```diff
@@ -40,5 +40,7 @@ def ssp_ssn_decryption_func(value, **kwargs):
 def zero_pad(digits):
     """Zero pad a string."""
     def transform(value, **kwargs):
```
Comment (elipe17) [inline tdrs-backend/tdpservice/parsers/transforms.py:42]: Can we add an extra unit test for this that tests the `None` case?

**Review by elipe17 — CHANGES_REQUESTED** [review]
I think Andy added testing for the `zero_pad` function. Just want to add the `None` case to that unit test

**Review by andrew-jameson — APPROVED** [review]

File: tdrs-backend/tdpservice/parsers/transforms.py (hunk 1 above)
Comment (raftmsohani) [reply tdrs-backend/tdpservice/parsers/transforms.py:42]: The test above is test for None value.

**Review by elipe17 — APPROVED** [review]

**Comment by ADPennington** [comment]
@raftmsohani can you check the unit tests on this PR when you get a chance?

**Comment by raftmsohani** [comment]
> @raftmsohani can you check the unit tests on this PR when you get a chance?

All good now!

**Review by ADPennington — APPROVED** [review]
this looks good to me @raftmsohani 🚀
